plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLUtils.java

```diff
@@ -1207,4 +1207,57 @@ public static String identifierToCanonicalForm(
         actualIdentifierString = forceUnquotted ? unquottedIdentifier : dialect.getQuotedIdentifier(unquottedIdentifier, true, false);
         return actualIdentifierString;
     }
+
+    public static String compact(String sql, SQLDialect sqlDialect) {
+
+        String res = removeComments(sql, sqlDialect);
+
+        final String dl = getScriptLineDelimiter(sqlDialect);
+        final String ls = GeneralUtils.getDefaultLineSeparator();
+
+        StringBuilder buffer = new StringBuilder();
+        List<String> result = new ArrayList<>();
+
+        String[] lines = res.split("\\R");
+
+        for (String line : lines) {
+            if (line.trim().isEmpty()) {
+                if (!buffer.isEmpty()) {
+                    result.add(buffer.toString().strip());
+                    buffer.setLength(0);
+                }
+                result.add("");
+                continue;
+            }
+
+            String trimmed = line.strip();
+
+            buffer.append(trimmed);
+            if (trimmed.endsWith(dl)) {
+                result.add(buffer.toString().strip());
+                buffer.setLength(0);
+            } else {
+                buffer.append(" ");
+            }
+        }
+
+        if (!buffer.isEmpty()) {
+            result.add(buffer.toString().strip());
+        }
+
+        return String.join(ls, result);
+
+    }
+
+    public static String removeComments(String sql, SQLDialect sqlDialect) {
```

Comment:
We have `org.jkiss.dbeaver.model.sql.SQLUtils#stripComments`, how is this function different?